classes/ReassuranceActivity.php

```diff
@@ -150,9 +150,12 @@ public static function getAllBlock()
      */
     public static function getAllBlockByStatus($id_lang = 1)
     {
+        $id_shop = (int) Context::getContext()->shop->id;
+
         $sql = 'SELECT * FROM `' . _DB_PREFIX_ . 'psreassurance` pr
             LEFT JOIN ' . _DB_PREFIX_ . 'psreassurance_lang prl ON (pr.id_psreassurance = prl.id_psreassurance)
             WHERE prl.id_lang = "' . (int) $id_lang . '"
+                AND prl.id_shop = "' . (int) $id_shop . '"
```

Comment:
This line is just a small part of what [PR 336 - Remove broken multishop support](https://github.com/PrestaShop/blockreassurance/pull/336/files#diff-3f34d294ae195f56ad0db4e3398c52d0356f388dc62595ee3d572e495fbbfa54L182) removed since March 2022. If we dive deeper in that PR changes, we can see to bring this line back we need to bring [its prerequisite line](https://github.com/PrestaShop/blockreassurance/pull/336/files#diff-8473374a31fcc2071f982bbe46a4f5d279c23ed5e4acdb0471cefeaf9121b7b8L163) back too.
For a whole view to make sure we won't bring the old feature back with known issues, I suggest reading the whole PR336 changes and conversation.